homeassistant/components/apcupsd/sensor.py

```diff
@@ -454,18 +538,38 @@ def infer_unit(value):
 class APCUPSdSensor(SensorEntity):
     """Representation of a sensor entity for APCUPSd status values."""
 
-    def __init__(self, data, description: SensorEntityDescription):
+    def __init__(
+        self,
+        data_service: APCUPSdData,
+        description: SensorEntityDescription,
+        enabled_by_default: bool,
+    ) -> None:
         """Initialize the sensor."""
+        # Set up unique id and device info if serial number is available.
+        if (serial_no := data_service.serial_no) is not None:
+            self._attr_unique_id = f"{serial_no}_{description.key}"
+            self._attr_device_info = {
+                "model": data_service.model,
+                "manufacturer": "APC",
+                "hw_version": data_service.hw_version,
+                "sw_version": data_service.sw_version,
+            }
```

Comment:
```suggestion
            self._attr_device_info = DeviceInfo(
                model=data_service.model,
                manufacturer="APC",
                hw_version=data_service.hw_version,
                sw_version=data_service.sw_version,
            )
```

Reply:
I recommend setting a name here and following our entity naming guidelines:

https://developers.home-assistant.io/docs/core/entity#entity-naming

Reply:
We have already set the `entity_description` in this entity, which contains a `name` field. From reading https://github.com/home-assistant/core/blob/dev/homeassistant/helpers/entity.py#L323-L324, it seems that it will fall back to using the `entity_description.name` if a name is not explicitly set.

I'm a bit confused here as to which is recommended, actually.  I can get rid of `entity_description` if it is discouraged (since I didn't see it mentioned in https://developers.home-assistant.io/docs/core/entity)

Also, all the names in current entity description objects contain a `UPS` prefix, I assume this is also discouraged from reading the doc you referenced, but I didn't change them due to compatibility. Do you think we should bring a breaking change for this?

Reply:
Since there's no unique_id set for the entities before this PR we should not change names to the new style now but do that after two releases when we remove the import of the YAML config.